main.go

```diff
@@ -0,0 +1,209 @@
+			fixed, err := splitArguments(argument)
+			if err != nil {
+				return nil, err
+
+			}
+			fixedArguments = append(fixedArguments, fixed[0], fixed[1])
+		} else {
+			fixedArguments = append(fixedArguments, argument)
+		}
+	}
+	return fixedArguments, nil
+}
+
+// Takes the input arguments and outputs them as separate values
+// parameterName = "Parent.child1.child2. ... .childn"
+// [parameterName-1 value-1, ...., parameterName-n value-n ] -> [[parent-1, child-1, ...child-n, value1],...]
+func parseArgs(args []string) [][]string {
+	var splitParameters [][]string
+	argsLength := len(args)
+	for index, parameterNames := range args[1:] {
+		if index%2 == 0 || index+2 == argsLength {
+			continue
+		}
+		parameterAndValue := strings.Split(strings.TrimPrefix(parameterNames, "--"), ".")
+		parameterAndValue = append(parameterAndValue, args[index+2])
+		splitParameters = append(splitParameters, parameterAndValue)
+	}
+	return splitParameters
+}
+
+// Iterates over all parameters and tries to set them
+func decorateType(serviceType runtime.Object, fieldNames [][]string) (runtime.Object, error) {
+	for _, v := range fieldNames {
+		_, err := setParameter(serviceType, v[:len(v)-1], v[len(v)-1])
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return serviceType, nil
+}
+
+// Iterates through service Type to find and set the required parameters
+// logs error and exits if parameter does not match any in Type specified field names
+func setParameter(serviceType runtime.Object, parameterHierarchy []string, value string) (runtime.Object, error) {
+	reflectedServiceType := reflect.ValueOf(serviceType).Elem()
+
+	for _, parameterName := range parameterHierarchy {
+		parameterName = strcase.ToCamel(parameterName)
+
+		if !reflectedServiceType.FieldByName(parameterName).IsValid() {
+			originalParameterName := parameterName
+			var err error
+			parameterName, err = fixStringCase(structs.Names(reflectedServiceType.Interface()), parameterName)
+			if err != nil {
+				err = fmt.Errorf("%w\n%s contains field with name %s : %t",
+					err,
+					reflectedServiceType.Type().Name(),
+					parameterName,
+					reflectedServiceType.FieldByName(parameterName).IsValid(),
+				)
+				return nil, err
+			}
+			info := fmt.Sprintf(
+				"Converted input parameter %s to %s to match correct fieldName\n",
+				originalParameterName,
+				parameterName,
+			)
+			logrus.Info(info)
+		}
+
+		if reflectedServiceType.FieldByName(parameterName).Kind() == reflect.Struct {
+			reflectedServiceType = reflectedServiceType.FieldByName(parameterName)
+		} else if reflectedServiceType.FieldByName(parameterName).Kind() == reflect.String {
+			reflectedServiceType = reflectedServiceType.FieldByName(parameterName)
+			reflectedServiceType.SetString(value)
+			fmt.Printf("setting\n-----------------------\nfield: %s\nvalue: %s\n-----------------------\n", strings.Join(parameterHierarchy, "."), value)
+		} else if reflectedServiceType.FieldByName(parameterName).Kind() == reflect.Int {
```

Comment:
This check should be changed to check of Int, Int8, Int16, Int32, Int64 ("Int" is equal to 2 and "Int64" is equal to 6, thus you can also do a range check 2-6).

Additionally we should probably support unsigned integers Uint, Unit8, Unit16, Uint32 and Uint64 (range 7-11). This needs an additional "if" statement because the setter function is called "SetUint".

Reply:
Added proper checking and support for lnts Uints and Floats.
Resolved by commit [f91af1f](https://github.com/vshn/appcat-cli/pull/1/commits/f91af1fa086ab4d0727af11c659096d1946339fd)